suncalc.js

```diff
@@ -300,11 +300,79 @@ SunCalc.getMoonTimes = function (date, lat, lng, inUTC) {
 
     return result;
 };
+              
+function calcMoonTransit(rize, set) {
+    if (rize > set) {
+        return new Date(set.getTime() + (rize - set) / 2);
+    } else {
+        return new Date(rize.getTime() + (set - rize) / 2);
+    }
+}
 
+function mainMoonTransit(rise, set, lat, lng) {
+    var main = "",
+        day = set.getDate(),
+        tempTransitBefore,
+        tempTransitAfter;
+
+    if ((rise && set) && (rise < set)) {
+        main = calcMoonTransit(rise, set);
+    } else {
+        if (rise) {
+            tempTransitAfter = calcMoonTransit(rise, SunCalc.getMoonTimes(new Date(rise).setDate(day + 1), lat, lng).set);
+            if (tempTransitAfter.getDate() === day) {
+                main = tempTransitAfter;
+            }
+        }
+
+        if (set) {
+            tempTransitBefore = calcMoonTransit(set, SunCalc.getMoonTimes(new Date(set).setDate(day - 1), lat, lng).rise);
+            if (tempTransitBefore.getDate() === day) {
+                main = tempTransitBefore;
+            }
+        }
+    }
+    return main;
+}
+
+function invertMoonTransit(rise, set, lat, lng) {
+    var invert = "",
+        day = set.getDate(),
+        tempTransitBefore,
+        tempTransitAfter;
+
+    if ((rise && set) && (set < rise)) {
+        invert = calcMoonTransit(rise, set);
+    } else {
+        if (rise) {
+            tempTransitBefore = calcMoonTransit(rise, SunCalc.getMoonTimes(new Date(rise).setDate(day - 1), lat, lng).set);
+            if (tempTransitBefore.getDate() === day) {
+                invert = tempTransitBefore;
+            }
+        }
+
+        if (set) {
+            tempTransitAfter = calcMoonTransit(set, SunCalc.getMoonTimes(new Date(set).setDate(day + 1), lat, lng).rise);
+            if (tempTransitAfter.getDate() === day) {
+                invert = tempTransitAfter;
+            }
+        }
+    }
```

Comment:
I feel like all this code could be DRY-ed up to be much shorter. There are minimal differences between normal and invert functions.